helix-term/src/commands.rs

```diff
@@ -5480,13 +5481,18 @@ fn increment_impl(cx: &mut Context, amount: i64) {
     let text = doc.text();
 
     let changes = selection.ranges().iter().filter_map(|range| {
-        let incrementor = NumberIncrementor::from_range(text.slice(..), *range)?;
-        let new_text = incrementor.incremented_text(amount);
-        Some((
-            incrementor.range.from(),
-            incrementor.range.to(),
-            Some(new_text),
-        ))
+        let incrementor: Option<Box<dyn Increment>> = if let Some(incrementor) =
+            DateTimeIncrementor::from_range(text.slice(..), *range)
+        {
+            Some(Box::new(incrementor))
+        } else if let Some(incrementor) = NumberIncrementor::from_range(text.slice(..), *range) {
+            Some(Box::new(incrementor))
+        } else {
+            None
+        };
+
+        let (range, new_text) = incrementor?.increment(amount);
+        Some((range.from(), range.to(), Some(new_text)))
```

Comment:
```suggestion
        let incrementor: Box<dyn Increment> = if let Some(incrementor) =
            DateTimeIncrementor::from_range(text.slice(..), *range)
        {
            Box::new(incrementor)
        } else if let Some(incrementor) = NumberIncrementor::from_range(text.slice(..), *range) {
            Box::new(incrementor)
        } else {
            return None;
        };

        let (range, new_text) = incrementor.increment(amount);
        Some((range.from(), range.to(), Some(new_text)))
```
I think this might be better?

Reply:
I agree.